static/app/components/replays/replayPlayer.tsx

```diff
@@ -80,8 +80,25 @@ function BasePlayerRoot({className, overlayContent, isPreview = false}: Props) {
 
   useVideoSizeLogger({videoDimensions, windowDimensions});
 
-  // Create the `rrweb` instance which creates an iframe inside `viewEl`
-  useEffect(() => initRoot(viewEl.current), [initRoot]);
+  // Sets the parent element where the player
+  // instance will use as root (i.e. where it will
+  // create an iframe)
+  useEffect(() => {
+    // XXX: This is smelly, but without the
+    // dependence on `isFetching` here, will result
+    // in ReplayContext creating a new Replayer
+    // instance before events are hydrated. This
+    // resulted in the `recording(Start/End)Frame`
+    // as the only two events when we instanciated
+    // Replayer and the rrweb Replayer requires all
+    // events to be present when instanciated.
```

Comment:
The culprit here is `useReplayData`, i lost my console logs but:

```
    const fetching =
      isFetchingReplay ||
      isFetchingAttachments ||
      isFetchingErrors ||
      isFetchingExtraErrors ||
      isFetchingPlatformErrors;
```

what happens is that we fetch two of these initially and in parallel (isFetchingReplay, isFetchingErrors), then, isFetchingAttachments and isFetchingExtraErrors only starts after isFetchingErrors is completed. But in between, we will have a point where all of them are false

| isFetchingReplay | isFetchingErrors | isFetchingAttachments | isFetchingExtraErrors | |
| - | - | - | - | - | 
| true | true | false | false | starting to fetch |
| false | false | false | false | first requests finished but the dependent fetches have not updated state yet |
| false | false | true | true | second batch starts |
| false | false | false | false | all requests finish |

When it hits row 2, a `useEffect` would trigger thinking fetching has been completed, and ReplayReader would return 2 synthetic events (replay.start and another one))